services/visual_recognition/v3.js

```diff
@@ -0,0 +1,364 @@
+          msg.result = {};
+          msg.result['error_code']= err.code;
+          if (!err.error)
+            msg.result['error']= err.error;
+          console.log('Error:', err.code);
+          node.error('Error code : ' +  err.code);
+          return;
+        }
+        else {
+          msg.result = 'Deleted classifier_id';
+          msg.payload='see msg.result';
+          node.send(msg); 
+          node.status({});
+        }
+      }; // actionCompleteDeleteClassifier
+      
+      // If the input is an image, need to stream the input in, giving time for the
+      // data to arrive, before invoking the service. 
+      node.status({fill:'blue', shape:'dot', text:'Calling '+ feature + ' ...'});
+      if (imageCheck(msg.payload)) {
+        temp.open({suffix: '.' + fileType(msg.payload).ext}, function (err, info) {
+          if (err) {
+            this.status({fill:'red', shape:'ring', text:'unable to open image stream'});          
+            message ='Node has been unable to open the image stream'; 
+            node.error(message, msg);
+            return;        
+          }  
+          stream_buffer(info.path, msg.payload, function () {
+          params['images_file'] = fs.createReadStream(info.path);
+          if ( msg.params != null && msg.params.classifier_ids != null)
+            params['classifier_ids']=msg.params['classifier_ids'];
+          if ( msg.params != null && msg.params.owners != null)
+            params['owners']=msg.params['owners'];
+          if ( msg.params != null && msg.params.threshold != null)
+            params['threshold']=msg.params['threshold'];
+          performAction(params, feature, actionComplete);
+          });
+        }); // temp
+      } else if (feature==='createClassifier') {   
+          var list_params = {};
+          var asyncTasks = [];
+          var prop = null;
+          for (var k in msg.params)
+          {
+            prop = k;
+            if (prop.indexOf('_examples')>=0)
+            {
+              // before pushing the function into the task array wrap the push 
+              // in an IIFE function, passing in the 'prop' parameter
+              (function(prop, list_params, msg) {
+
+               asyncTasks.push(function (callback) {
+                  var buffer = msg.params[prop];
+                  temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+                    if (err) {
+                      this.status({fill:'red', shape:'ring', 
+                                   text:'unable to open image stream'});          
+                      var message ='Node has been unable to open the image stream'; 
+                      node.error(message, msg);
+                      callback('open error on '+prop);
+                    }  
+                    stream_buffer(info.path, msg.params[prop], function () {
+                      list_params[prop]=fs.createReadStream(info.path);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'list_params' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2690779421/issues/source?bid=3330694&fileBranchId=3339165#l294)